src/averageAges.js

```diff
@@ -55,7 +84,30 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const allMothersNames = people.map(person => person.mother);
+  const listedMothers = people.filter(person =>
+    allMothersNames.includes(person.name));
+  const listedMothersNames = listedMothers.map(person => person.name);
+  let children = people.filter(person =>
+    listedMothersNames.includes(person.mother));
+
+  children.map(child => {
+    const mother = listedMothers.find(person => person.name === child.mother);
+
+    if (mother.name === child.mother) {
+      child.motherBorn = mother.born;
+    }
+
+    return child;
+  });
+
+  children = onlyWithSon
+    ? children.filter(child => child.sex === 'm')
+    : children;
+
+  const diff = children.map(child => child.born - child.motherBorn);
+
+  return diff.reduce((a, b) => a + b) / diff.length;
```

Comment:
Need more descriptive names and you can put this into a separate function as reuse it in all three functions